src/test/java/com/google/sps/GetImageServletTest.java

```diff
@@ -0,0 +1,62 @@
+package com.google.sps;
+
+import static org.mockito.Mockito.when;
+
+import com.google.appengine.tools.development.testing.LocalBlobstoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.google.sps.servlets.GetImageServlet;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+// Responsible for testing the GetImageServlet to make sure that the image
+// being properly served.
+@RunWith(JUnit4.class)
+public final class GetImageServletTest {
+  private final LocalServiceTestHelper helper =
+      new LocalServiceTestHelper(new LocalBlobstoreServiceTestConfig());
+  private static GetImageServlet getImageServlet;
+  private static final String BLOB_KEY_PARAMETER = "blobkey";
+  private static final String TEST_KEY = "M-kUqkjD_Cr--tf0Ga26vg";
+  @Mock private HttpServletRequest request;
+  @Mock private HttpServletResponse response;
+
+  @Before
+  public void setUp() {
+    helper.setUp();
+    MockitoAnnotations.initMocks(this);
+    this.getImageServlet = this.newGetImageServlet();
+  }
+
+  // Creating local get servlet
+  public GetImageServlet newGetImageServlet() {
+    GetImageServlet getImageServlet = new GetImageServlet();
+    return getImageServlet;
+  }
+
+  @Test
+  public void doGetGetImageTest() throws IOException {
```

Comment:
Add tests that exercise:

1. Not uploading an image
2. Uploading an empty image
3. Retrieving an image while logged in
4. Retrieving an image while not logged in
5. Uploading an empty image and then retrieving it